pkg/metrics/metrics.go

```diff
@@ -0,0 +1,41 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	log = logf.Log.WithName("metrics")
+)
+
+const (
+	MetricNamespace = "nsx"
+	MetricSubsystem = "operator"
+	HealthKey       = "health"
+	ScrapeTimeout   = 30
+)
+
+var (
+	NSXOperatorHealthStats = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: MetricNamespace,
+			Subsystem: MetricSubsystem,
+			Name:      HealthKey,
+			Help:      "Last health status for NSX-Operator. 1 for 'status' label with current status.",
+		},
+		[]string{"status"},
+	)
+)
+
+var registerMetrics sync.Once
```

Comment:
For `sync.Once`, I'm afraid we cannot invoke Register multiple times for different metrics. e.g.
```
metrics.Register(metrics.NSXOperatorHealthStats)
metrics.Register(metrics.NSXOperatorControllerSyncTotal)
metrics.Register(metrics.NSXOperatorControllerUpdateTotal)
...
```
will fail, right?

Reply:
What error do you encounter with the above? I did not test it with multiple metrics though.
I checked the doc and think we can set it 
func InitializePrometheusMetrics() {
	log.Info("Initializing prometheus metrics")
	Register(NSXOperatorHealthStats, NSXOperatorControllerSyncTotal, NSXOperatorControllerUpdateTotal)
}

Reply:
got it, thanks.